libs/zephyr-agent/src/tests/ze-agent.spec.ts

```diff
@@ -132,16 +136,98 @@ runner('ZeAgent', () => {
         `npx nx run sample-webpack-application:build --skip-nx-cache --verbose`,
       ].join(' ');
       await exec(cmd);
-      const deployResultUrls = await _getAppTagUrls(application_uid);
+
+      const deployResultUrls = await _getAppTagUrls(uuid);
       expect(deployResultUrls).toBeTruthy();
+
       for (const url of deployResultUrls) {
         expect(url).toBeTruthy();
         const content = await _fetchContent(url);
         const match = content.match(/<title>([^<]+)<\/title>/);
         expect(match).toBeTruthy();
         expect(match?.[1]).toEqual('SampleReactApp');
       }
-      await _cleanUp(application_uid);
+      await _cleanUp(uuid);
+    },
+    integrationTestTimeout
+  );
+
+  it(
+    'should deploy Rspack',
+    async () => {
+      const app = 'sample-rspack-application';
+      const uuid = `${app}.${appProject}.${appOrg}`;
+
+      const envs = [
+        `ZE_IS_PREVIEW=${ZE_IS_PREVIEW()}`,
+        `ZE_API=${ZEPHYR_API_ENDPOINT()}`,
+        `ZE_API_GATE=${dev_api_gate_url}`,
+        `ZE_SECRET_TOKEN=${getSecretToken()}`,
+        `DEBUG=zephyr:*`,
+      ];
+
+      const cmd = [
+        'npx cross-env',
+        ...envs,
+        `npx nx run sample-rspack-application:build --skip-nx-cache --verbose`,
+      ].join(' ');
+      await exec(cmd);
+
+      // Verify deployment
+      const deployResultUrls = await _getAppTagUrls(uuid);
+      expect(deployResultUrls).toBeTruthy();
+
+      // Check each deployed URL
+      for (const url of deployResultUrls) {
+        expect(url).toBeTruthy();
+        const content = await _fetchContent(url);
+        const match = content.match(/<title>([^<]+)<\/title>/);
```

Comment:
at the end this should be a matrix with not only content but headers check as well
to actually verity deployment you need to inject buildid or something similar into html content